modules/jaggery-apps/appmgt/src/modules/application/get/list.jag

```diff
@@ -1283,4 +1283,107 @@ var getRepoUserInfo = function(applicationKey, artifacts) {
        }
        return isApplicationsOfUserChanged;
     }
-%>
+    
+    var getEsbArtifacts = function(applicationKey, version) {
+        var log = new Log();
+        var applicationInfo = modApplication.getAppInfo(applicationKey);
+    
+        // get the list of artifacts deployed on the esb
+        var ws = require('ws');
+        var getCarApps = new ws.WSRequest();
+        var options = new Array();
+        options.useSOAP = 1.1;
+        options.action = "urn:getSynapseAppData";
+        modManager.setAuthHeaders(options);
+        var depoyedCarName = applicationKey + "_" + version;
+        var payload = '<syn:getSynapseAppData xmlns:syn="http://synapse.mgt.application.carbon.wso2.org"> <syn:appName xmlns:syn="http://synapse.mgt.application.carbon.wso2.org">' + depoyedCarName + '</syn:appName> </syn:getSynapseAppData>';
+        var result;
+        var carApp;
+        var carAppName = [];
+        var proxyServices = [];
+        var sequences = [];
+        try {
+            getCarApps.open(options, "https://192.168.19.51:30013/services/SynapseApplicationAdmin", false);
+            getCarApps.send(payload);
+            result = getCarApps.responseE4X;
+            carApp = result.*::["return"];
+
+           for(var i=0;i<carApp.length();i++){
+                carAppName.push(carApp[i].*::appName);
+                proxyServices.push(carApp[i].*::proxyServices);
+                sequences.push(carApp[i].*::sequences);
+           }
+
+        } catch (e) {
+            log.error(e.toString());
```

Comment:
This is wrong!. Log the error(as log.error(e)) with a separate error message.
On the other hand, since we are not throwing an error, how do we know error occurred in the backend.